CMakeLists.txt

```diff
@@ -33,10 +33,17 @@ if(ENABLE_TESTING)
   enable_testing()
 endif()
 
-# suppress warnings
+# build flags
 
 add_compile_options(
   $<$<AND:$<STREQUAL:$<COMPILE_LANGUAGE>,CXX>,$<CXX_COMPILER_ID:AppleClang>>:-Wno-deprecated-declarations>
+  $<$<AND:$<STREQUAL:$<COMPILE_LANGUAGE>,CXX>,$<CXX_COMPILER_ID:AppleClang>,$<VERSION_LESS:$<CXX_COMPILER_VERSION>,9.3>>:-Werror>
+  $<$<AND:$<STREQUAL:$<COMPILE_LANGUAGE>,CXX>,$<CXX_COMPILER_ID:AppleClang>,$<VERSION_LESS:$<CXX_COMPILER_VERSION>,9.3>>:-Wall>
+  $<$<AND:$<STREQUAL:$<COMPILE_LANGUAGE>,CXX>,$<CXX_COMPILER_ID:AppleClang>,$<VERSION_LESS:$<CXX_COMPILER_VERSION>,9.3>>:-Wextra>
```

Comment:
Is there a reason you limit the warnings to certain compiler versions?

Reply:
New compiler versions normally add new warnings to -Wall -Wextra. My rationale is that the current Travis CI uses two specific compilers and especially GCC 4.8 is rather old. So we can only ensure for these two compilers that we don't have any warnings. If we would unconditionally switch on -Wall ... with -Werror a user of this library who might use a newer GCC or Clang version, may run into the problem that he cannot compile this library anymore.

Do you know of a better solution?

Reply:
If we guard the flags for the CI then the version number can be removed.

Reply:
done. I removed the no-deprecation-declarations. Is it ok?